messaging/src/main/java/org/axonframework/eventhandling/ReplayToken.java

```diff
@@ -77,16 +169,24 @@ public static TrackingToken createReplayToken(TrackingToken tokenAtReset) {
         return createReplayToken(tokenAtReset, null);
     }
 
+
     /**
-     * Indicates whether the given message is "redelivered", as a result of a previous reset. If {@code true}, this
-     * means this message has been delivered to this processor before its token was reset.
+     * Extracts the context from a {@code message} of the matching {@code contextClass}.
+     * <p>
+     * Will resolve to an empty {@code Optional} if the provided token is not a {@link ReplayToken} or the context an
+     * instance of the provided {@code contextClass}.
      *
-     * @param message The message to inspect
-     * @return {@code true} if the message is a replay
+     * @param message      The message to extract the context from
+     * @param contextClass The class the context should match
+     * @param <T>          The type of the context
+     * @return The context, if the provided token is a {@link ReplayToken} and the context an instance of the provided
+     * {@code contextClass}.
      */
-    public static boolean isReplay(Message<?> message) {
-        return message instanceof TrackedEventMessage
-                && isReplay(((TrackedEventMessage) message).trackingToken());
+    public static <T> Optional<T> replayContext(Message<?> message, @Nonnull Class<T> contextClass) {
```

Comment:
Why allow any `Message`? Why not only `EventMessage` implementations?

Reply:
This was because the `ReplayContextParameterResolverFactory` also matches on the generic `Message`. Changing it because the other was changed as well.